### PR TITLE
Add alpine docker image definition

### PR DESCRIPTION
**Pull request by vvmruder**
Just a suggestion

### CODE REVIEW

**Comment by vvmruder** [comment]
@balazsdukai From my point of view this should work now. I tried to squeeze it as much as possible. But couldn't get image smaller than 1.4gb :cry: 
You might try it now. It lasts a while to build locally. But it includes spatialite support (I had some chat with someone I know... and now it works) even with proj6 and gdal 3. Iam specially interested how it works for your flow using PostGIS. So: Let me know.

**Comment by vvmruder** [comment]
Since there is already a travis configuration I was not editing it. But maybe we should also update this like in the other projects.

**Comment by vvmruder** [comment]
And: I will update the readme soon.

**Comment by balazsdukai** [comment]
Great, thanks! I'm gonna test it and merge. I'll also update the travis build script I think, it'll be useful.
Yeah I guess 1,4 gb it is. The size makes sense I think since it includes the whole OSGeo stack of libraries :-) .

**Comment by vvmruder** [comment]
Yes. I tried to offer as much input as possible for vector features.

You are aware of the "master" version of spatialite? Iam convinced it will work for the moment. But still, it is master...

**Comment by balazsdukai** [comment]
Hm no, what's the "master" version of spatialite? :-)

**Comment by vvmruder** [comment]
Its basically the current state of development. So it potentially contains pre alpha code. But as far as I understood the guys maintaining spatialite are really careful about working master branch. So in the moment this should be ok.

**Comment by balazsdukai** [comment]
@vvmruder I'm building the image but I get a connection time out for download.osgeo. Did you ever got that?

```bash
(17/18) Installing unzip (6.0-r4)

(18/18) Installing .proj4-deps (20191107.131458)

Executing busybox-1.30.1-r2.trigger

OK: 174 MiB in 40 packages

Connecting to download.osgeo.org (140.211.15.30:80)

wget: can't connect to remote host (140.211.15.30): Operation timed out

Error: ResponseItem.ErrorDetail[code=1,message=The command '/bin/sh -c apk --update add sqlite libstdc++ sqlite-libs libgcc &&     apk --update add --virtual .proj4-deps         make         gcc         g++         file         sqlite-dev         unzip &&     cd /tmp &&     wget http://download.osgeo.org/proj/proj-${PROJ_VERSION}.tar.gz &&     tar xfvz proj-${PROJ_VERSION}.tar.gz &&     rm -f proj-${PROJ_VERSION}.tar.gz &&     wget http://download.osgeo.org/proj/proj-datumgrid-1.8.zip &&     unzip proj-datumgrid-1.8.zip -d proj-${PROJ_VERSION}/nad/ &&     rm -f proj-datumgrid-1.8.zip &&     wget http://download.osgeo.org/proj/proj-datumgrid-europe-1.1.zip &&     unzip proj-datumgrid-europe-1.1.zip -d proj-${PROJ_VERSION}/nad/ &&     rm -f proj-datumgrid-europe-1.1.zip &&     wget http://download.osgeo.org/proj/proj-datumgrid-north-america-1.1.zip &&     unzip proj-datumgrid-north-america-1.1.zip -d proj-${PROJ_VERSION}/nad/ &&     rm -f proj-datumgrid-north-america-1.1.zip &&     wget http://download.osgeo.org/proj/proj-datumgrid-oceania-1.0.zip &&     unzip proj-datumgrid-oceania-1.0.zip -d proj-${PROJ_VERSION}/nad/ &&     rm -f proj-datumgrid-oceania-1.0.zip &&     cd proj-${PROJ_VERSION} &&     ./configure &&     make &&     make install &&     echo "Entering root folder" &&     cd / &&    echo "Cleaning dependencies tmp and manuals" &&     apk del .proj4-deps &&     rm -rf /tmp/* &&     rm -rf /user/local/man &&     for i in /usr/local/lib/libproj*; do strip -s $i 2>/dev/null || /bin/true; done &&     for i in /usr/local/lib/geod*; do strip -s $i 2>/dev/null || /bin/true; done &&     for i in /usr/local/bin/proj*; do strip -s $i 2>/dev/null || /bin/true; done &&     proj' returned a non-zero code: 1]
```

**Comment by vvmruder** [comment]
Tried to reach it in browser. But it seems really lagging. So probably they have some problem with their servers?

**Comment by balazsdukai** [comment]
Yes, seems so.

**Comment by balazsdukai** [comment]
I also get a warning about Boost when building 3dfier. Do you get this @vvmruder ?

```bash
CMake Warning at /usr/share/cmake/Modules/FindBoost.cmake:905 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindBoost.cmake:1027 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/share/cmake/Modules/FindBoost.cmake:1727 (_Boost_MISSING_DEPENDENCIES)
  CMakeLists.txt:24 (find_package)


CMake Warning at /usr/share/cmake/Modules/FindBoost.cmake:905 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindBoost.cmake:1027 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/share/cmake/Modules/FindBoost.cmake:1727 (_Boost_MISSING_DEPENDENCIES)
  CMakeLists.txt:24 (find_package)


CMake Warning at /usr/share/cmake/Modules/FindBoost.cmake:905 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindBoost.cmake:1027 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/share/cmake/Modules/FindBoost.cmake:1727 (_Boost_MISSING_DEPENDENCIES)
  CMakeLists.txt:24 (find_package)


CMake Warning at /usr/share/cmake/Modules/FindBoost.cmake:905 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindBoost.cmake:1027 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/share/cmake/Modules/FindBoost.cmake:1727 (_Boost_MISSING_DEPENDENCIES)
  CMakeLists.txt:24 (find_package)


CMake Warning at /usr/share/cmake/Modules/FindBoost.cmake:905 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindBoost.cmake:1027 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/share/cmake/Modules/FindBoost.cmake:1727 (_Boost_MISSING_DEPENDENCIES)
  CMakeLists.txt:24 (find_package)


-- Boost version: 1.71.0
```

**Comment by vvmruder** [comment]
Yes. This warnings I had too. I was intentionally ignoring them because whole processes run. But I had to use this new boost version because of proj6. This is a topic we can discuss. I thought it might be a good idea to go with some up to date versions. But maybe we should stick to proj4. This would make use of 'older' versions of involved libraries possible.

**Comment by balazsdukai** [comment]
The process runs for me too, as least seemingly. However, 3dfier doesn't generate any output. It says it read the point clouds, but I'm pretty sure it didn't, because 3dfier finishes waay to quickly. Did you get that issue?

`docker run --rm -v "$(pwd)":/data tudelft3d/3dfier:latest 3dfier testarea_config_unix.yml --CityJSON testarea.json`

**Comment by vvmruder** [comment]
I dar to ask. But: Are the files defined as full access on the host machine? Since we use random user it need to be configured like this. And I recognized that 3dfier fails silently if there are permission problems with the configured files.

Which OS you are using?

**Comment by vvmruder** [comment]
image

I always give this permission pattern on all folders and files recursively to make 3dfier working correctly.

**Comment by balazsdukai** [comment]
But `read` should be sufficient, no?

```bash
-rw-rw-r-- 1 balazs balazs  3532883 jul 27  2018 ahn3_cropped_1.laz
-rw-rw-r-- 1 balazs balazs  5836325 jul 20  2018 ahn3_cropped_2.las
-rw-rw-r-- 1 balazs balazs  1170107 jul 27  2018 ahn3_cropped_2.laz
```

**Comment by vvmruder** [comment]
That's strange. Are you using it with PostGIS? What is the output of 3dfier?

**Comment by balazsdukai** [comment]
Sooory, my bad :-). I forgot that the working directory was write protected, so 3dfier was not allowed to write the output.

But indeed, 3dfier fails silently in such chase.

**Comment by vvmruder** [comment]
Maybe we could improve this. It is especially disturbing if all process runs through and in the end when it tries to write the output like CityJSON it fails silently and you are left with no output. After all the time of worried waiting :)

**Comment by balazsdukai** [comment]
Indeed, feel free to add your thoughts to https://github.com/tudelft3d/3dfier/issues/95

**Comment by vvmruder** [comment]
So process is running as expected now? With PostGIS?

**Comment by balazsdukai** [comment]
Unfortunately no. It seems that GDAL was not compiled with the Postgres driver, or the build didn't work out the way it was supposed to. In case of a successful build one should get the following output:

```bash
$ ogrinfo --formats | grep 'PostgreSQL -vector-'
  PostgreSQL -vector- (rw+): PostgreSQL/PostGIS
```

But I get nothing for:

```bash
docker run --rm -it tudelft3d/3dfier:latest ogrinfo --formats | grep 'PostgreSQL -vector-'
```

**Comment by vvmruder** [comment]
Oh ... 

I used this https://github.com/tudelft3d/3dfier/pull/88/files#diff-3254677a7917c6c01f55212f86c57fbfR376 to enable it while building. But it seems not to work. So need some time to investigate. Do you found out something meanwhile?

**Comment by balazsdukai** [comment]
No, I haven't looked into this further.

**Comment by vvmruder** [comment]
I've updated it to work as expected. Could you please try it again?

File: Dockerfile [hunk 1]
```diff
@@ -433,6 +433,7 @@ RUN ln -s /usr/local/lib64/libCGAL.so.13 /usr/local/lib && \
         cmake \
         linux-headers && \
     cd /tmp && \
+    rm -rf build && \
```
Comment (vvmruder) [inline Dockerfile]: Is this necessary? Did you have issues there? Normally all `/tmp` is cleared after each build stage.

**Comment by vvmruder** [comment]
I will.

@balazsdukai Did you read my comment on commit?

File: Dockerfile (hunk 1 above)
Comment (balazsdukai) [reply Dockerfile]: Yes, but that's a different thing.
Here we copy the whole source directory, and if the source directory contains a `build` directory (often the case), then `mkdir` fails.

https://github.com/vvmruder/3dfier/blob/a18af880e4e2bfda571b7168b72c030851432187/Dockerfile#L416

Comment (vvmruder) [reply Dockerfile]: Ok i agree. But I added it in .dockerignore https://github.com/tudelft3d/3dfier/pull/88/files#diff-f7c5b4068637e2def526f9bbc7200c4eR1
So it should be omitted on the copy command from docker file. Isn't it the case?

Comment (balazsdukai) [reply Dockerfile]: Hm I see. It still copied the `build` for me...
Oh wait :man_facepalming: 
I didn't copy `.dockerignore` when I switched branch :grimacing: 
I'll try again.

Comment (balazsdukai) [reply Dockerfile]: Alrite, `.dockerignore` is indeed doing what it's meant to.

**Pull request merged**